lib/client/identityfile/identity.go

```diff
@@ -614,10 +614,14 @@ func KeyFromIdentityFile(identityPath, proxyHost, clusterName string) (*client.K
 // is also added to its profile store with the limited profile data available
 // in the identity file.
 //
-// Since identity files do not save a proxy address, proxyAddr must be provided
-// to fill in this data gap. clusterName can also be provided to aim the key at
-// a leaf cluster rather than the default root cluster.
+// Use [proxyAddr] to specify the host:port-like address of the proxy.
+// This is necessary because identity files do not store the proxy address.
+// Additionally, the [clusterName] argument can ve used to target a leaf cluster
+// rather than the default root cluster.
 func NewClientStoreFromIdentityFile(identityFile, proxyAddr, clusterName string) (*client.Store, error) {
+	if proxyAddr == "" {
+		return nil, trace.BadParameter("missing proxyAddr loading identityFile. Login, configure a proxy address, or pass --proxy on the command line.")
```

Comment:
Is the same function called when using `tctl` with the identity file? Because if I think the error message might be incorrect since tctl's flag is `--auth-server`, not `--proxy`.

Reply:
What I'm finding is that when called with an identity file but no proxy, `tctl` calls down to this `NewClientStoreFromIdentityFile` function with an empty (`""`) `proxyAddr`, which `utils.Host()` fails to parse with an even-more-confusing error message to the end user.

I was originally thinking of putting this check in here only, but since it really seems like an argument-parsing requirement, I moved the conditional check up to `Run()` where the rest of the command line argument parsing is happening.

But, then I'm also left thinking that this function doesn't really work with an empty `proxyAddr`, and opted to leave this check in here to leave a better error message in case something calls this with a useless/non-Host-ey `proxyAddr` value.

Reply:
Yeah, I get that, so what kind of error do you get when you execute `tctl` with only `--identity` flag?

Reply:
The before and after is in the PR description and the bug that prompted this.

If we only call `tsh` with `--identity`, it prints this confusing error message (since the real error is that there is no proxy set, and it is trying to parse `""` as a hostname):

```
% ~/src/teleport/build/tsh --identity=key.pem --insecure ssh root@localhost
ERROR: missing parameter hostname
```

Reply:
Ah, I see what you mean now -- I was glossing over `tctl` vs. `tsh`

I still get the same confusing error message, though with a different call stack that also goes via `NewClientStoreFromIdentityFile`:
```
% ~/src/teleport/build/tctl  --identity=~/teleport/config/jof.pem users ls
ERROR: missing parameter hostname
```

I suppose I may as well try and address this for `tctl` as well as `tsh`.